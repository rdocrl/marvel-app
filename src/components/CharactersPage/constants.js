export const characterFilterOptions = [
  { id: 'nameStartsWith', label: 'Name' },
  { id: 'comics', label: 'Comic' },
  { id: 'stories', label: 'Story' }
];
