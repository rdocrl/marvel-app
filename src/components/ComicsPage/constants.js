export const comicFilterOptions = [
  { id: 'titleStartsWith', label: 'Title' },
  { id: 'issn', label: 'Issue Number' }
];
