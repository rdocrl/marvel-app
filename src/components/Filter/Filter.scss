.filter {
  color: gray;
  font-size: 0.8rem;

  &__types {
    display: flex;
    gap: 4px;
  }

  &__text {
    margin-top: 8px;
    background: white;
    border-radius: 3px;
    height: 2rem;
    display: flex;
    padding: 4px;
    gap: 6px;
    border: 1px solid rgb(240, 240, 240);

    input {
      border: none;

      &:focus {
        outline: none;
        color: gray;
      }
    }
  }
}
