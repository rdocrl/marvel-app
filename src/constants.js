export const resourceTypes = {
  COMIC: 'comics',
  CHARACTER: 'characters',
  STORY: 'stories'
};
