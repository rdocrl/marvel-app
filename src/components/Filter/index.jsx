import { forwardRef } from 'react';
import search from '../../assets/search.png';
import './Filter.scss';

const Filter = forwardRef(({ options, onTypeChange, onTextChange, selected, text }, ref) => {
  const handleTypeChange = (e) => {
    const { value } = e.target;
    onTypeChange(value);
  };

  const handleTextChange = (e) => {
    const { value } = e.target;
    onTextChange(value);
  };

  return (
    <section className="filter">
      <div className="filter__types">
        {options.map(({ id, label }) => (
          <div key={id}>
            <input
              type="radio"
              value={id}
              checked={selected === id}
              onChange={handleTypeChange}
              id={id}
            />
            <label htmlFor={id}>{label}</label>
          </div>
        ))}
      </div>
      <div className="filter__text">
        <img src={search} />
        <input value={text} placeholder="Search" onChange={handleTextChange} ref={ref} />
      </div>
    </section>
  );
});

Filter.displayName = 'Filter';

export default Filter;
