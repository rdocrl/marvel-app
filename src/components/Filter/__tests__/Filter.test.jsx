import { fireEvent, render, screen } from '@testing-library/react';
import Filter from '../';
import { filterOptions } from '../../../constants';

describe('<Filter /> tests', () => {
  describe('Snapshots', () => {
    it('Should render', () => {
      const { asFragment } = render(<Filter options={filterOptions.characters} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Functional', () => {
    it('Should call onTypeChange', () => {
      const onTypeChangeMock = jest.fn();
      render(<Filter options={filterOptions.characters} onTypeChange={onTypeChangeMock} />);
      fireEvent.click(screen.getByLabelText('Comic'));
      expect(onTypeChangeMock).toBeCalled();
    });

    it('Should call onTextChange', () => {
      const onTextChangeMock = jest.fn();
      render(<Filter options={filterOptions.characters} onTextChange={onTextChangeMock} />);
      const input = screen.getByRole('textbox');
      fireEvent.change(input, { target: { value: 'Spider' } });
      expect(onTextChangeMock).toBeCalled();
    });
  });
});
