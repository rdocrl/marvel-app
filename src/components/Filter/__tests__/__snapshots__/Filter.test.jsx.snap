// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> tests Snapshots Should render 1`] = `
<DocumentFragment>
  <section
    class="filter"
  >
    <div
      class="filter__types"
    >
      <div>
        <input
          id="nameStartsWith"
          type="radio"
          value="nameStartsWith"
        />
        <label
          for="nameStartsWith"
        >
          Name
        </label>
      </div>
      <div>
        <input
          id="comics"
          type="radio"
          value="comics"
        />
        <label
          for="comics"
        >
          Comic
        </label>
      </div>
      <div>
        <input
          id="stories"
          type="radio"
          value="stories"
        />
        <label
          for="stories"
        >
          Story
        </label>
      </div>
    </div>
    <div
      class="filter__text"
    >
      <img
        src="search.png"
      />
      <input
        placeholder="Search"
        value=""
      />
    </div>
  </section>
</DocumentFragment>
`;
